src/uct/base/uct_mem.c

```diff
@@ -129,6 +130,30 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
             }
             break;
 
+        case UCT_ALLOC_METHOD_THP:
+#ifdef MADV_HUGEPAGE
+            if (!ucs_is_thp_enabled()) {
+                break;
+            }
+
+            alloc_length = ucs_align_up(min_length, ucs_get_huge_page_size());
+            address = ucs_memalign(ucs_get_huge_page_size(), alloc_length
+                                   UCS_MEMTRACK_VAL);
+            if (address != NULL) {
+                status = madvise(address, alloc_length, MADV_HUGEPAGE);
+                if (status != UCS_OK) {
+                    ucs_error("madvise failure status (%d) errno(%s)",
+                              status, strerror(errno));
+                    ucs_free(address);
+                } else {
+                    goto allocated_without_md;
+                }
+            }
+#endif
+
+            ucs_debug("failed to allocate %zu with thp method", alloc_length);
```

Comment:
if MADV_HUGEPAGE is not defined , it would always print bogus debug message "failed to allocate..."